combo-demo/player/Player.gd

```diff
@@ -0,0 +1,100 @@
+extends KinematicBody2D
+
+enum States { MOVE, LIGHT_ATTACK, HEAVY_ATTACK }
+
+const DRAG_FACTOR := 15.0
+const SPEED_MOVE := 400.0
+const SPEED_ATTACK := 1000.0
+
+var _state = States.MOVE
+var _accept_next_attack := true
+var _direction := Vector2.ZERO
+var _velocity := Vector2.ZERO
+
+var _previous_light_combo := "light_combo_3"
+var _previous_heavy_combo := "heavy_combo_2"
```

Comment:
This isn't great. I'd rather have something a bit more of a system that makes editing combos efficient. An array of attacks you used as part of a combo would be preferable